packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1060,6 +1060,31 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   ///  * [ModalBarrier], the widget that implements this feature.
   String get barrierLabel;
 
+  /// The tween that is used for animating the modal barrier in and out.
+  ///
+  /// The modal barrier is the scrim that is rendered behind each route, which
+  /// generally prevents the user from interacting with the route below the
+  /// current route, and normally partially obscures such routes.
+  ///
+  /// For example, when a dialog is on the screen, the page below the dialog is
+  /// usually darkened by the modal barrier.
+  ///
+  /// While the route is animating into position, the color is animated from
+  /// transparent to the specified color.
+  ///
+  /// If this getter would ever start returning a different color,
+  /// [changedInternalState] should be invoked so that the change can take
+  /// effect.
+  ///
+  /// It defaults to a [CurveTween] with a [Curves.ease] curve.
+  ///
+  /// See also:
+  ///
+  ///  * [barrierColor], which determines the color that the modal transitions
+  ///    to.
+  ///  * [AnimatedModalBarrier], the widget that implements this feature.
+  Animatable<double> get barrierTween => CurveTween(curve: Curves.ease);
```

Comment:
This is kinda odd since the full barrierTween would really also include the ColorTween? I wonder what you would use this except for changing the curve? Should we just make the curve customizable? Or are there other use cases?

Reply:
It becomes sort of weird since currently, the internal `ColorTween`  uses the `barrierColor` as an input. If we have `barrierTween` be the entire customizable curve, then it could theoretically ignore `barrierColor`. An example would be:
```dart
PopupRoute(
  barrierTween: ColorTween(
    begin: _kTransparent,
    end: Colors.orange, // no dependence on barrierColor
  ).chain(CurveTween(curve: Curves.ease));
  // ...
);
``` 
Would that be worth making the entire curve customizable?

Reply:
I was thinking we limit this even more. Instead of allowing you to customize the CurveTween you can only customize the Curve used by that CurveTween.